### PR TITLE
ci: add tubular in ci checks

### LINKED ISSUE #1202: ci: add downstream tests for tubular
looks like tubular are now using Narwhals 🥳 

https://github.com/lvgig/tubular/pull/328

let's add a downstream test for them (if it's not too taxing)? https://github.com/narwhals-dev/narwhals/blob/main/.github/workflows/downstream_tests.yml

### PR DESCRIPTION
**Pull request by FBruzzesi**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [x] 🐳 Other

## Related issues 

- Closes #1202 

## Checklist

- [ ] Code follows style guide (ruff)
- [ ] Tests added 
- [ ] Documented the changes

### CODE REVIEW

**Comment by FBruzzesi** [comment]
Can't replicate the issue locally even with fresh new env. (link to failing [action](https://github.com/narwhals-dev/narwhals/actions/runs/11391931290/job/31696817589?pr=1211))

Error:

> INTERNALERROR> pytest.PytestConfigWarning: Unknown config option: env

**Comment by DeaMariaLeon** [comment]
@FBruzzesi  Would that have something to do this line in Narwhals' `pyproject.toml`? :
`env = [
    "MODIN_ENGINE=python",
]`

If so, should that environment variable be added? (to the tubular test section on `.github/workflows/downstream_tests.yml` that is being added).. Asking for a friend 🥸

**Comment by FBruzzesi** [comment]
> @FBruzzesi Would that have something to do with this line in Narwhals' `pyproject.toml`? : `env = [ "MODIN_ENGINE=python", ]`
> 
> If so, should that environment variable be added? (to the tubular test section on `.github/workflows/downstream_tests.yml` that is being added).. Asking for a friend 🥸

I wonder why none of the other downstream test ends up with similar issue if that was the case! I will try to add the env variable (and pytest-env)

**Comment by DeaMariaLeon** [comment]
It went further!..

**Comment by FBruzzesi** [comment]
> It went further!..

Yep yep, that was a good catch Dea, thanks for spotting it! Next issue ahead 😁🙈

File: .github/workflows/downstream_tests.yml [hunk 1]
```diff
@@ -7,6 +7,9 @@ on:
 
 jobs:
   altair:
+    env:
+      UV_SYSTEM_PYTHON: true
```
Comment (FBruzzesi) [inline .github/workflows/downstream_tests.yml:11]: I find this cleaner and easier to not forget down the line. Description from docs:

> UV_SYSTEM_PYTHON: Equivalent to the --system command-line argument. If set to true, uv will use the first Python interpreter found in the system PATH. 
> WARNING: UV_SYSTEM_PYTHON=true is intended for use in continuous integration (CI) or containerized environments and should be used with caution, as modifying the system Python can lead to unexpected behavior.

**Comment by FBruzzesi** [comment]
Regarding marimo typecheck failing, seems we are missing some libraries?

File: .github/workflows/downstream_tests.yml (hunk 1 above)
Comment (DeaMariaLeon) [reply .github/workflows/downstream_tests.yml:11]: If this is set `UV_SYSTEM_PYTHON=true`, then:`MODIN_ENGINE=python` is not needed?

Comment (FBruzzesi) [reply .github/workflows/downstream_tests.yml:11]: Redirecting pytest to the tubular pyproject.toml solved all the issues for tubular (`pytest tests --config-file=pyproject.toml`) 

`UV_SYSTEM_PYTHON=true` allows us to not specify `--system` each time when using `uv` to install dependencies

Comment (FBruzzesi) [reply .github/workflows/downstream_tests.yml]: This was making marimo ci fail. I rolled back all the changes except for adding tubular in downstream test and readme.

**Comment by DeaMariaLeon** [comment]
I'm not sure if requesting me as a reviewer was an accident or a github bug 😆, but LGTM.

**Review by MarcoGorelli — APPROVED** [review]
thanks @FBruzzesi and @DeaMariaLeon !

**Pull request merged**